install.sh

```diff
@@ -337,6 +343,33 @@ setup_env() {
         fi
     fi
 
+    if [ "${CATTLE_AGENT_UNINSTALL_LOCAL}" = "true" ]; then
+        if [ -z "${CATTLE_AGENT_UNINSTALL_LOCAL_LOCATION}" ]; then
+            fatal "No local uninstall location was specified"
+        fi
+        UNINSTALL_SOURCE=local
+    else
+        UNINSTALL_SOURCE=remote
+        if [ -z "${CATTLE_AGENT_UNINSTALL_URL}" ] && [ -n "${CATTLE_AGENT_BINARY_BASE_URL}" ]; then
+            CATTLE_AGENT_UNINSTALL_URL="${CATTLE_AGENT_BINARY_BASE_URL}/uninstall.sh"
+        fi
+
+        if [ -z "${CATTLE_AGENT_UNINSTALL_URL}" ]; then
+            if [ $(curl --connect-timeout 60 --max-time 60 -s https://api.github.com/rate_limit | grep '"rate":' -A 4 | grep '"remaining":' | sed -E 's/.*"[^"]+": (.*),/\1/') = 0 ]; then
```

Comment:
Can we add an if-check on the `VERSION` so that if it is already set above, then we don't set it again here?

Also a nit: the version setting here and above is the same. Can this be extracted to its own function? It may not be because of uninstall versus install.